shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1272,65 +1284,220 @@ - (void)updateViewportPadding {
-  // in the screen to see if the keyboard is visible.
-  if (CGRectIntersectsRect(keyboardFrame, screenRect)) {
-    CGFloat bottom = CGRectGetHeight(keyboardFrame);
-    CGFloat scale = [UIScreen mainScreen].scale;
-    // The keyboard is treated as an inset since we want to effectively reduce the window size by
-    // the keyboard height. The Dart side will compute a value accounting for the keyboard-consuming
-    // bottom padding.
-    self.targetViewInsetBottom = bottom * scale;
-  } else {
-    self.targetViewInsetBottom = 0;
-  }
   [self startKeyBoardAnimation:duration];
 }
 
-- (void)keyboardWillBeHidden:(NSNotification*)notification {
-  NSDictionary* info = [notification userInfo];
+- (BOOL)shouldIgnoreKeyboardNotification:(NSNotification*)notification
+                        notificationName:(NSNotificationName)notificationName {
+  BOOL isKeyboardNotificationForThisView = [self isKeyboardNotificationForThisView:notification];
+  BOOL isKeyboardRotated = [self isKeyboardRotated:notification];
+
+  // Don't ignore UIKeyboardWillHideNotification notifications.
+  if (notificationName == UIKeyboardWillHideNotification) {
+    // Skip hide notification when rotation in progress unless triggered by another app.
+    if (isKeyboardRotated && isKeyboardNotificationForThisView) {
+      return YES;
+    }
+    return NO;
+  }
+
+  // Ignore notification when keyboard's dimensions and position are all zeroes,
+  // for UIKeyboardWillChangeFrameNotification. This happens when keyboard is dragged.
+  NSDictionary* info = notification.userInfo;
+  CGRect keyboardFrame = [[info objectForKey:UIKeyboardFrameEndUserInfoKey] CGRectValue];
+  if (notificationName == UIKeyboardWillChangeFrameNotification &&
+      CGRectEqualToRect(keyboardFrame, CGRectZero)) {
+    return YES;
+  }
+
+  // Don't ignore other times a keyboard's height or width is 0.
+  if (CGRectIsEmpty(keyboardFrame)) {
+    return NO;
+  }
 
   // Ignore keyboard notifications related to other apps.
+  if (!isKeyboardNotificationForThisView) {
+    return YES;
+  }
+
+  // Ignore notification when keyboard is in process of being rotated.
+  if (isKeyboardRotated) {
+    return YES;
+  }
+
+  return NO;
+}
+
+- (BOOL)isKeyboardNotificationForThisView:(NSNotification*)notification {
+  NSDictionary* info = notification.userInfo;
+  // Keyboard notifications related to other apps.
   id isLocal = info[UIKeyboardIsLocalUserInfoKey];
   if (isLocal && ![isLocal boolValue]) {
-    return;
+    return NO;
   }
-
-  // Ignore keyboard notifications if engine’s viewController is not current viewController.
+  // Engine’s viewController is not current viewController.
   if ([_engine.get() viewController] != self) {
-    return;
+    return NO;
   }
+  return YES;
+}
+
+- (BOOL)isKeyboardRotated:(NSNotification*)notification {
+  // When the keyboard's width at the beginning of the animation equals the screen's
+  // current height, we can assume the keyboard was rotated.
```

Comment:
"was rotated" or "is the process of being rotated"?

Reply:
So these notifications are right before it rotates. So perhaps "will rotate" is the better phrase. I'll update the comments and function name.